Include seasonal gear in Market class lists

SabreCat opened the pull request:
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

**Previously**, the gear section of the regular Market did not display seasonal class equipment. This proved a bit confusing, as users would expect to see currently buyable, class-specific gear there.
**Now**, Gold-buyable seasonal gear shows up in the Market under the expected class heading.

**Previously**, some special items did not show up in the Market if they were broken due to running out of Health. This affected, for example, "legendary" quest items and Take This challenge gear.
**Now**, these "classless" special items do show up in the Market. This is a partial fix for #9731.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

SabreCat committed:
fix(shops): include seasonal gear in Market class lists

negue commented on website/common/script/libs/shops.js:
```diff
@@ -155,7 +155,16 @@ shops.getMarketGearCategories = function getMarketGear (user, language) {
+          class: classType,
+        },
+      };
+      if (gearItem.specialClass === classType) return gearItem.canOwn(classShift);
```
this should also merge the `specialGear` https://github.com/HabitRPG/habitica/blob/c1bd762ff159112e5b735b387f3500af94c60280/website/common/script/libs/shops.js#L175
which adds the special gear that has been already owned


once both merged it should also check about the current season gear. Cause I don't think we should list gear that can't be purchased at the moment (which also never been owned)

SabreCat replied:
OK, do you suggest chaining (or following up with) a Lodash `.merge` calling that function then?

This won't list gear that can't be purchased. `canOwn` takes care of that part.

SabreCat committed:
fix(market): display non-seasonal broken special items
Also fixes a bug where if a current seasonal item was broken, it would show up twice.
